recipes/cocoyaxi/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class CocoyaxiConan(ConanFile):
+    name = "cocoyaxi"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/idealvin/cocoyaxi"
+    license = "MIT"
+    description = "A go-style coroutine library in C++11 and more."
+    topics = ("cocoyaxi", "coroutine", "c++11")
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libcurl": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libcurl": False,
+        "with_openssl": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_libcurl:
+            self.requires("libcurl/7.79.1")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1l")
```

Comment:
```suggestion
        if self.options.with_libcurl or self.options.with_openssl:
            self.requires("openssl/1.1.1m")
```

according to your CMakeLists